nap/utils.py

```diff
@@ -60,19 +51,21 @@ def flatten_params(k, vs):
 
         # since we can have more than one value for a single key, we use a
         # tuple of two tuples instead of a dictionary
-        params_tuple = tuple(itertools.chain(*flat_params))
-        param_string = urllib.parse.urlencode(params_tuple)
+        params_tuple = tuple(sorted(itertools.chain(*flat_params), key=lambda key_value: key_value[0]))
```

Comment:
We don't want them to always be in the same order, we want to preserve the order of what could be interpreted as an ordered list.